Makefile

```diff
@@ -0,0 +1,89 @@
+VERSION := 0.19.1
+
+# install directory layout
+PREFIX ?= /usr/local
+INCLUDEDIR ?= $(PREFIX)/include
+LIBDIR ?= $(PREFIX)/lib
+PCLIBDIR ?= $(LIBDIR)/pkgconfig
+
+# collect C++ sources, and link if necessary
+CPPSRC := $(wildcard src/*.cc)
+
+ifeq (, $(CPPSRC))
+	ADDITIONALLIBS := 
+else
+	ADDITIONALLIBS := -lc++
+endif
+
+# collect sources
+ifneq ($(AMALGAMATED),1)
+	SRC := $(wildcard src/*.c)
+	SRC += $(CPPSRC)
+	
+	# do not double-include amalgamation
+	SRC := $(filter-out src/lib.c,$(SRC))
+else
+	# use amalgamated build
+	SRC := src/lib.c
+endif
+OBJ := $(addsuffix .o,$(basename $(SRC)))
+
+# ABI versioning
+SONAME_MAJOR := 0
+SONAME_MINOR := 0
+
+# define default flags, and override to append mandatory flags
+CFLAGS ?= -O3 -Wall -Wextra -Werror
+override CFLAGS += -std=gnu99 -fPIC -Ilib/src -Ilib/include
```

Comment:
For this I'd suggest a different variable (e.g. `TREE_SITTER_CFLAGS`), also set via `?=`, that you'd add to the `$(CC)` calls down below.  That way callers still have the ability to change the if they need to.  (It's just that it's much less likely that they should / need to than the stuff in `CFLAGS`.)

Reply:
I started working on your suggestion, and realized an issue. The `$(CC)` calls were not happening, due to default make rules and file patterns not matching. Again, I am most definitely not a make expert.

It also would mean this build process would be different in a subtle way from the runtime's. Is that what we want?

Reply:
I would actually suggest removing the `build/%.o: %.c` rules entirely, and using Make's default rules for creating `.o` files. The `.o` files would just get put in the `src` folder by default instead of `build`, but I think that's ok.

Reply:
Done! But, the flag control is still potentially an issue. I've at least corrected it so the runtime's behavior applies for both c and c++ files though.